arrow-core/src/main/kotlin/arrow/core/extensions/either/zip/EitherZip.kt

```diff
@@ -0,0 +1,40 @@
+package arrow.core.extensions.either.zip
+
+import arrow.core.Either
+import arrow.core.Tuple2
+
+@JvmName("zip")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated(
+  "@extension projected functions are deprecated",
+  ReplaceWith(
+    "this.zip(arg1, ::Tuple2)",
+    "arrow.core.zip", "arrow.core.Tuple2"
+  ),
+  DeprecationLevel.WARNING
+)
+fun <A, B, E> Either<E, A>.zip(arg1: Either<E, B>): Either<E, Tuple2<A, B>> =
```

Comment:
Things that return `Tuple2` should return `Pair` instead. Same for `Tuple3`, they should return `Triple`